### PR TITLE
feat: update Input component to Diete v5

### PR DESCRIPTION
**Pull request by hlavkat**

### CODE REVIEW

File: src/lib/components/ui/input.svelte [hunk 1]
```diff
@@ -1,396 +1,402 @@
...
+			</label>
+		{/if}
+		{#if controls && type === 'number'}
+			<div class="control-buttons">
+				<button class="substract" onclick={() => (inputValue -= 1)}
+					><Subtract size={dimension === 'small' ? 16 : 24} /></button
+				>
+				<button class="add" onclick={() => (inputValue += 1)}
+					><Add size={dimension === 'small' ? 16 : 24} /></button
+				>
```
Comment (vojtechsimetka) [inline src/lib/components/ui/input.svelte:76]: I believe these should be secondary buttons, should have a focus state and that focus state should remove the focus state from the input.

**Review by vojtechsimetka — CHANGES_REQUESTED** [review]
Looks good! Just one remark about the control buttons in the number input.

That being said, we should discuss if the control buttons  (`+` & `-`) should not be `Snippets` that are passed to the component instead. That means we can pass in any `Button` and the logic would be outside of the Input component. It would make the input more generic (we could have any button there). Not sure if this is what we want though @agazso @david-gauquelin any thoughts?

**Comment by agazso** [comment]
> That being said, we should discuss if the control buttons (`+` & `-`) should not be `Snippets` that are passed to the component instead. That means we can pass in any `Button` and the logic would be outside of the Input component. It would make the input more generic (we could have any button there). Not sure if this is what we want though @agazso @david-gauquelin any thoughts?

Considering there are at least 10 different types of textual input fields in [HTML alone](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input) (number, tel, search, url etc), I would focus on just to have a simple text input field that can be later extended with functionality (buttons, validation etc.). Or if that makes it too complex, just keep it as a simple text input and later we can add more input types (for example numeric).

**Review by vojtechsimetka — APPROVED** [review]
LGTM on my end. For now the control buttons are not configurable but everything is ready for it to be provided with some `Snippet`.

**Comment by vojtechsimetka** [comment]
Should now work in Firefox correctly. Thanks for the link!

**Review by agazso — APPROVED** [review]

**Comment by david-gauquelin** [comment]
A bit late to the party sorry, but I would like to add my thoughts here and explained how I thought about this when designing them.

I tried to think about a flexible system, so in my view the input field is generic and the buttons are basically just separate components attached to it. This means that they can be focused, and behave just as the other buttons. By the way this is how it's built in Figma, these are nested instances of the existing buttons with some minor changes on the stroke and rounded corners in order to be attached properly to the input.

I used a similar system for 3 other input components: search (with delete and optional search button), date (with calendar button) and color picker (with palette button).

So these are a few examples that we can build "by default" but on top, I plan to document this "aggregating" system in order to allow more combinations. Potentially even beyond the inputs, maybe we can aggregate other type of components together. I have to explore this soon as I will start working on "Layout components".

**Pull request merged**